src/utils.rs

```diff
@@ -258,15 +259,31 @@ fn internal_exec_cmd(cmd: &str, args: &[&str]) -> Option<CommandOutput> {
             log::trace!("Using {:?} as {:?}", full_path, cmd);
             full_path
         }
-        Err(e) => {
-            log::trace!("Unable to find {:?} in PATH, {:?}", cmd, e);
+        Err(error) => {
+            log::trace!("Unable to find {:?} in PATH, {:?}", cmd, error);
             return None;
         }
     };
 
+    let time_limit = Duration::from_millis(500);
+
     let start = Instant::now();
-    match Command::new(full_path).args(args).output() {
-        Ok(output) => {
+
+    let process = match Command::new(full_path)
+        .args(args)
+        .stderr(Stdio::piped())
+        .stdout(Stdio::piped())
+        .spawn()
+    {
+        Ok(process) => process,
+        Err(error) => {
+            log::info!("Unable to run {:?}, {:?}", cmd, error);
+            return None;
+        }
+    };
+
+    match process.with_output_timeout(time_limit).wait() {
```

Comment:
I think terminating the process on timeout makes sense.

```suggestion
    match process.with_output_timeout(time_limit).terminating().wait() {
```

Reply:
I'm not sure about this the [process_control docs](https://docs.rs/process_control/3.0.0/process_control/struct.Terminator.html#safety) seem to indicate that `terminating` isn't safe to use in production.

Reply:
[terminat**ing**](https://docs.rs/process_control/3.0.0/process_control/trait.Timeout.html#tymethod.terminating) should be safe. [It's supposed what to be one of the things that sets it apart from other crates.](https://docs.rs/process_control/3.0.0/process_control/index.html#comparable-crates)

If you don't want to add `terminating` the crate isn't required for the timeout functionality.

```rust
let (sender, receiver) = channel();

thread::spawn(move|| {
    sender.send(process.output()).unwrap();
});

receiver.recv_timeout(duration)
```

Reply:
Ah nice I didn't see that.